src/content/docs/en/guides/backend/supabase.mdx

```diff
@@ -3,10 +3,806 @@ title: Supabase & Astro
+                    placeholder="Password"
+                />
+                <input
+                    class="border rounded px-3 py-2"
+                    aria-live="polite"
+                    name="confirm-password"
+                    type="password"
+                    placeholder="Confirm Password"
+                />
+                <button
+                    class="block px-3 py-2 w-full text-white bg-gray-800 border rounded hover:border-2"
+                    aria-live="polite"
+                    type="submit"
+                >
+                    Sign Up
+                </button>
+                <p class="text-center text-gray-800">Already have an account? <a href="/signin">Sign In</a></p>
+
+                <p class="text-center text-gray-800">Or Sign In with</p>
+                <div class="flex justify-center space-x-4">
+                    <button id="github" class="bg-gray-800 text-white px-3 py-2 rounded github">Github</button>
+                </div>
+            </form>
+        </div>
+    </div>        
+</Layout>
+
+<script>
+    import { signUp, signInWithGithub } from "../lib/auth";
+
+    const form = document.querySelector("form")
+
+    form?.addEventListener("submit", async (e) => {
+        e.preventDefault();
+        console.log("submitted")
+        const formData = new FormData(e.target as HTMLFormElement);
+        const creds = {
+            email: formData.get("email") as string,
+            password: formData.get("password") as string,
+            confirmPassword: formData.get("confirm-password") as string,
+        };
+        
+        if (creds.password !== creds.confirmPassword) {
+            console.log("Passwords do not match");
+            alert("Passwords do not match");
+            return;
+        } else {
+            try {
+                await signUp(creds.email, creds.password);
+            } catch (error) {
+                console.log(error);
+            }
+        }
+    });
+
+    const github = document.getElementById("github");
+    github?.addEventListener("click", async (e) => {
+        e.preventDefault();
+        await signInWithGithub();
+    })
+</script>
+```
+
+### Profile
+
+1. Go to `src/pages`
+2. Create a new file named `profile.astro`
+3. Add the following code to `profile.astro`:
+
+```astro
+---
+// src/pages/profile.astro
+import Layout from "../layouts/Layout.astro";
+import { getSessionData, setSessionData, AuthState} from "../lib/auth";
+
+let accessToken = Astro.cookies.get("sb-access-token").value
+let refreshToken = Astro.cookies.get("sb-refresh-token").value
+
+console.log("accessToken", accessToken);
+console.log("refreshToken", refreshToken);
+const session = await setSessionData(accessToken, refreshToken);
+
+console.log("session", session);
+
+if (session.data.session === null || (accessToken === undefined && refreshToken === undefined)) {
+	return Astro.redirect("/signin");
+}
+
+console.log("Auth State", AuthState())
```

Comment:
I don't think we need to set up auth state since the session management is handled server side.

Reply:
Yes I figured it out later. I will fix it and update this guide soon.